### DISCUSSION TITLE
The ability to customize plots when using `TSDataset.plot` and functions from `etna.analysis`

### DISCUSSION THREAD

**Discussion by ArtemLiA** [etna-team/etna#446] [Improvements]
### Description

Hi there! How would you like to add the ability to setup up graphs more flexible during plotting them using `TSDataset.plot` method and the `etna.analysis` module functions? 

### Thoughts

This can be implemented, for example, by adding the `**kwargs` argument.

**Comment by d-a-bunin** [comment]
Hi! Could you elaborate on this? Where these `kwargs` should go to inside the plotting function? May be it can be more flexible to add parameters like `plot_params: Dict[str, Any]` which captures all the additional parameters for the `plt.plot` function, but it can be very dependent on other arguments of this `plt.plot`.
Can you give some more specific example?

**Reply by ArtemLiA** [reply]
I think using `**kwargs` for graphs customization would be more convenient because it would make it possible to pass parameters using `param_name=param_value` syntax. 

In the simplest case, for example, in the case of the `TSDataset.plot` method, `kwargs` could be directly passed to the `plt.plot` function. For more complex visualization functions we could use prefix notation for passing `kwargs` parameters. For example, we could pass to `etna.analysis.plot_forecast` arguments like `forecast_ls='-r'` or `train_linestyle='--'`.

**Reply by d-a-bunin** [reply]
For some methods there could be different set of additional parameters for different calls inside the function (not only plotting probably). Putting them all under `kwargs` will be inconvenient. Having some functions with `kwargs` some with `plot_params` could be inconsistent. Probably, it isn't important, but it could be.

Some parameters set this way could probably break the plots, and I'm not sure that we could check all such cases.

**Reply by ArtemLiA** [reply]
We don't have to give ability to pass parameters to all functions which are involved in plotting. For example, for the `etna.analysis.plot_forecast` function, we could pass kwargs only to functions outlining forecast_ts, train_ts and test_ts.

Regarding parameters checking: I think we don't need to check all parameters, we should only specify in the documentation what function they will be passed to.

In any case, my idea from the beginning was to add the possibility of wider customization of graphs rendered using `etna` tools. And it is not necessary to do this "globally" for all functions at once. This can be done "locally", that is, specific changes can be proposed for each function. For example, how would you like to add three arguments to the `etna.analysis.plot_forecast` function: `train_plot_params`, `test_plot_params`, `forecast_plot_params`, which would be used when drawing train, test and forecast, respectively?
